### PR TITLE
Support renderType in schema

### PR DESCRIPTION
**Pull request by 928PJY**
[AB#291280](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/291280)

Still depends on the changes of schema [AB#288448](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/288448)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6452)

### CODE REVIEW

File: src/docfx/build/toc/BuildTableOfContents.cs [hunk 1]
```diff
@@ -42,12 +42,9 @@ public static void Build(Context context, Document file)
             {
                 if (context.Config.OutputType == OutputType.Html)
                 {
-                    if (file.IsHtml)
```
Comment (yufeih) [inline src/docfx/build/toc/BuildTableOfContents.cs:45]: TOC is a component in docs-ui template but is content in pdf template, this if check is still needed then?

Comment (928PJY) [reply src/docfx/build/toc/BuildTableOfContents.cs:45]: TOC can always be able to generated to a HTML like `conceptual`, `landingData`, but whether to generate a HTML should be decided by the outputType

File: docs/specs/escape.yml [hunk 2]
```diff
@@ -142,6 +142,7 @@ inputs:
     Link to <xref:a%26b>
   _themes/ContentTemplate/schemas/TestData.schema.json: |
     {
+
```
Comment (yufeih) [inline docs/specs/escape.yml:145]: nit: no change?

File: src/docfx/build/document/DocumentProvider.cs [hunk 3]
```diff
@@ -229,7 +229,7 @@ private Document GetDocumentCore(FilePath path)
         {
             var contentType = _buildScope.GetContentType(path);
             var mime = _input.GetMime(contentType, path);
-            var isHtml = _templateEngine.IsHtml(contentType, mime.Value);
+            var isHtml = _templateEngine.IsHtml(contentType, mime);
```
Comment (yufeih) [inline src/docfx/build/document/DocumentProvider.cs:232]: Consider replace the term `IsHtml` with `RenderType`?

File: src/docfx/lib/schema/JsonSchema.cs [hunk 4]
```diff
@@ -41,6 +41,11 @@ internal class JsonSchema
         [JsonConverter(typeof(OneOrManyConverter))]
         public JsonSchemaType[]? Type { get; set; }
 
+        /// <summary>
+        /// Type of the render type.
+        /// </summary>
+        public RenderType RenderType { get; set; } = RenderType.Component;
```
Comment (yufeih) [inline src/docfx/lib/schema/JsonSchema.cs:47]: Is `Content` a better default?

File: src/docfx/template/TemplateEngine.cs [hunk 5]
```diff
@@ -161,6 +161,30 @@ public void Dispose()
             _js.Dispose();
         }
 
+        private bool IsContentScheme(SourceInfo<string?> mime)
```
Comment (mingwli) [inline src/docfx/template/TemplateEngine.cs:164]: Scheme => Schema

File: src/docfx/template/TemplateEngine.cs [hunk 6]
```diff
@@ -54,14 +54,14 @@ internal class TemplateEngine : IDisposable
...
+        public RenderType GetRenderType(ContentType contentType, SourceInfo<string?> mime)
         {
             return contentType switch
             {
-                ContentType.Redirection => true,
-                ContentType.Page => IsConceptual(mime) || IsLandingData(mime) || _mustacheTemplate.HasTemplate($"{mime}.html"),
-                ContentType.TableOfContents => _mustacheTemplate.HasTemplate($"toc.html"),
-                _ => false,
+                ContentType.Redirection => RenderType.Content,
+                ContentType.Page => (IsConceptual(mime) || IsLandingData(mime) || IsContentScheme(mime)) ? RenderType.Content : RenderType.Component,
```
Comment (yufeih) [inline src/docfx/template/TemplateEngine.cs:62]: Eliminate the conversion to bool `RenderType` --> `bool` --> `RenderType`?. If it is not conceptual or landing data, return `GetSchema().RenderType`?

```suggestion
                ContentType.Page => GetPageRenderType(mime)
```

**Review by yufeih — APPROVED** [review]

**Review by mingwli — APPROVED** [review]
:shipit:

**Pull request merged**